### PR TITLE
Improve HTML parser and recursive splitter

### PR DESCRIPTION
**Pull request by SayaZhang**
1. Improve HTML parser: fix text tag(p, span etc) parser and remove the `\n` in the text tag
2. Improve recursive splitter separators: add `.` as a separator for long text

### CODE REVIEW

File: uniflow/op/extract/split/recursive_character_splitter.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 class RecursiveCharacterSplitter(Op):
     """Recursive character splitter class."""
 
-    default_separators = ["\n\n", "\n", " ", ""]
+    default_separators = ["\n\n", "\n", ".", " ", ""]
```
Comment (goldmermaid) [inline uniflow/op/extract/split/recursive_character_splitter.py:14]: shall we use ". " rather than "."? since the later may separate decimals like "0.9".

Comment (goldmermaid) [reply uniflow/op/extract/split/recursive_character_splitter.py:14]: also do we need to add the "." back after the split?

Comment (CambioML) [reply uniflow/op/extract/split/recursive_character_splitter.py:14]: +1

Comment (SayaZhang) [reply uniflow/op/extract/split/recursive_character_splitter.py:14]: Make sense! `". "` is better. I think there is no need to add `"."` again.

**Review by CambioML — APPROVED** [review]

**Pull request merged**